lib/cabal-project-parser.nix

```diff
@@ -121,99 +121,6 @@ let
-            EOF
-
-            cabal v2-update ${name}
-            cp -r $HOME/.cabal/packages/${name} $out
-        ''
-        else evalPackages.runCommand name ({
-          nativeBuildInputs = [ cabal-install evalPackages.curl nix-tools ] ++ evalPackages.haskell-nix.cabal-issue-8352-workaround;
-          LOCALE_ARCHIVE = pkgs.lib.optionalString (evalPackages.stdenv.buildPlatform.libc == "glibc") "${evalPackages.glibcLocales}/lib/locale/locale-archive";
-          LANG = "en_US.UTF-8";
-          preferLocalBuild = true;
-        } // pkgs.lib.optionalAttrs (sha256 != null) {
-          outputHashMode = "recursive";
-          outputHashAlgo = "sha256";
-          outputHash = sha256;
-        }) ''
-            HOME=$(mktemp -d)
-            mkdir -p $HOME/.cabal/packages/${name}
-            cat <<EOF > $HOME/.cabal/config
-            repository ${name}
-              url: ${attrs.url}
-              ${pkgs.lib.optionalString (attrs ? secure) "secure: ${attrs.secure}"}
-              ${pkgs.lib.optionalString (attrs ? root-keys) "root-keys: ${attrs.root-keys}"}
-              ${pkgs.lib.optionalString (attrs ? key-threshold) "key-threshold: ${attrs.key-threshold}"}
-            EOF
-
-            export SSL_CERT_FILE=${evalPackages.cacert}/etc/ssl/certs/ca-bundle.crt
-            cabal v2-update ${name}
-            cp -r $HOME/.cabal/packages/${name} $out
-        '';
-      # Output of hackage-to-nix
-      hackage = import (
-        evalPackages.runCommand ("hackage-to-nix-" + name) {
-          nativeBuildInputs = [ cabal-install evalPackages.curl nix-tools ];
-          LOCALE_ARCHIVE = pkgs.lib.optionalString (evalPackages.stdenv.buildPlatform.libc == "glibc") "${evalPackages.glibcLocales}/lib/locale/locale-archive";
-          LANG = "en_US.UTF-8";
-          preferLocalBuild = true;
-        } ''
-          mkdir -p $out
-          hackage-to-nix $out ${repoContents}/01-index.tar ${attrs.url}
-        '');
-      # Directory to `lndir` when constructing a suitable $HOME/.cabal dir
-      repo = {
-        ${name} = repoContents;
-      };
-    };
-
-  parseRepositories = evalPackages: cabalProjectFileName: sha256map: inputMap: cabal-install: nix-tools: projectFile:
-    let
-      # This will leave the name of repository in the first line of each block
-      blocks = pkgs.lib.splitString "\nrepository " ("\n" + projectFile);
-      repoBlocks = builtins.map (parseRepositoryBlock evalPackages cabalProjectFileName sha256map inputMap cabal-install nix-tools) (pkgs.lib.lists.drop 1 blocks);
-    in {
-      extra-hackages = pkgs.lib.lists.map (block: block.hackage) repoBlocks;
-      repos = pkgs.lib.lists.foldl' (x: block: x // block.repo) {} repoBlocks;
-    };
-
 in {
-  inherit parseIndexState parseSourceRepositoryPackages parseRepositories
-    # These are only exposed for tests
-    parseSourceRepositoryPackageBlock parseRepositoryBlock;
+  inherit parseIndexState parseSourceRepositoryPackages;
```

Comment:
Can your tool also spit out at least the index-states? would that let us fix the problem with us parsing them wrong? I guess the s-r-ps are harder because we rely on the magic comments...